src/main/js/helpers/legend.js

```diff
@@ -205,11 +209,16 @@ const removeLegendItem = (legendSVG, dataTarget) =>
  * Only if showLegend is enabled.
  *
  * @private
+ * @param {object} config - Graph config object derived from input JSON
  * @param {object} container - d3 Container svg
  * @returns {object} - d3 svg object
  */
-const createLegend = (container) =>
-    container.append("ul").classed(styles.legend, true).attr("role", "list");
+const createLegend = (config, container) =>
+    container
+        .append("ul")
+        .classed(styles.legend, true)
+        .attr("role", "list")
+        .style("flex-direction", config.bindLegendTo && "column");
```

Comment:
What does the purpose of `flex-direction` here? Why did we add this change?

Reply:
When legendPadding is provided, we are removing the carbon default margin of 8px around the legend and changing the carbon default left and right padding of 8px each for the legend to whatever the consumer is providing. Reducing the padding and removing the margin is causing the legends to stack horizontally one after the other in the application like this: (here bindLegendTo is provided)

![image](https://user-images.githubusercontent.com/60426528/82419477-1bc9b100-9a9c-11ea-8ed2-45079af2e57b.png)

This is because carbon is setting flex-direction as row [here](https://github.com/cerner/carbon-graphs/blob/master/src/main/less/core.less#L91) which will stack the legends horizontally.

So when bindLegendTo is provided, since we want to legends to display one below the other, I have set the flex-direction to column which will stack the legends vertically.